integration/update_array_compat_test.go

```diff
@@ -238,3 +238,61 @@ func TestUpdateArrayCompatPullAll(t *testing.T) {
 
 	testUpdateCompat(t, testCases)
 }
+
+func TestUpdateArrayCompatPushEach(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]updateCompatTestCase{
+		"Document": {
+			update: bson.D{{"$push", bson.D{{"v", bson.D{
+				{"$each", bson.A{bson.D{{"field", int32(42)}}}},
+			}}}}},
+			skipForTigris: "Mixed types are not supported for Tigris.",
+		},
+		"String": {
+			update: bson.D{{"$push", bson.D{{"v", bson.D{{"$each", bson.A{"foo"}}}}}}},
+		},
+		"Int32": {
+			update: bson.D{{"$push", bson.D{{"v", bson.D{
+				{"$each", bson.A{int32(1), int32(42), int32(2)}},
+			}}}}},
+			skipForTigris: "Mixed types are not supported for Tigris.",
```

Comment:
I mentioned why `Int32` would not pass here https://github.com/FerretDB/FerretDB/pull/2187#discussion_r1135707528
 but it seems that I wasn't quite clear.
When you run `$each` with the array of integers against an array of doubles Tigris would convert those integers to doubles implicitly. MongoDB apart from that would just append integers. This would result in different responses between Tigris and MongoDB.